src/palace/manager/integration/patron_auth/sirsidynix_authentication_provider.py

```diff
@@ -220,17 +220,39 @@ def remote_patron_lookup(
         patrondata.complete = True
         fields: dict = data["fields"]
         patrondata.personal_name = fields.get("displayName")
-        patron_type: str = fields["patronType"].get("key", "")
-        patrondata.library_identifier = patron_type
+
+        if self.library_identifier_field == "patrontype":
+            patron_type: str = fields["patronType"].get("key", "")
+            patrondata.library_identifier = patron_type
+        elif self.library_identifier_field == "barcode":
+            pass
+            # do nothing sense the library identifier field and restriction type are handled in the authentication
+            # function (see enforce_library_identifier_restriction())
+        else:
+            # this should never happen
+            raise PalaceValueError(
+                f"Unexpected  library_identifier_field value ({self.library_identifier_field})"
+            )
 
         # Basic block reasons
 
-        if not fields.get("approved", False):
+        # Some Symphony installations appear to use "standing" to indicate approval.
+        # TODO we may want to make the approval mechanism configurable.
+        if not (
+            fields.get("approved", False)
+            or fields.get("standing", {}).get("key", "").lower() == "ok"
+        ):
             patrondata.block_reason = SirsiBlockReasons.NOT_APPROVED
             return patrondata
 
         # If the patron type ends with a disallowed suffix the
         # patron will be authenticated but marked as blocked.
+        if patron_type is not None:
+            for suffix in self.sirsi_disallowed_suffixes:
+                if patron_type.endswith(suffix):
+                    patrondata.block_reason = SirsiBlockReasons.PATRON_BLOCKED
+                    return patrondata
+
```

Comment:
It looks like the exact same loop is run just below this `if` block. It is run even if `patron_type` is None. I suspect that was meant to be deleted, but if the tests don't fail because of this, we are probably missing some test cases, too.